core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -89,7 +89,10 @@ class TopicConfigHandler(private val logManager: LogManager, kafkaConfig: KafkaC
       kafkaController.enableTopicUncleanLeaderElection(topic)
     }
 
-    Try(topicConfig.getProperty(KafkaConfig.MinInSyncReplicasProp).toInt) match {
+    // If the new topic config does not have min.insync.replicas configured, i.e. the topic shall use the default value,
+    // ensure that the controller knows about it.
+    val missingMinISRPropertyValue = "-1"
```

Comment:
Can this be a constant somewhere so that the code in the controller could read something like:

```scala
if (isValidReplicaCount(minInSyncReplicas)) {
      controllerContext.topicMinIsrConfig += topic -> minInSyncReplicas
} else if (minInsyncReplicas == REMOVED_PER_TOPIC_CONIFG) {
      controllerContext.topicMinIsrConfig -= topic
 } else {
      throw or deal with error
}
```

Reply:
Added `MissingPerTopicConfig` under `Defaults` and used it from both `ConfigHandler` and `KafkaController`.
Didn't add a sanity check (i.e. `isValidReplicaCount`) or dealt with the bad config values for `min.insync.replicas` here, because I think the validity of setting a minISR is irrelevant here.